sphinxcontrib/blockdiag.py

```diff
@@ -219,20 +219,41 @@ def html_depart_blockdiag(self, node):
 
 
 def get_image_format_for(builder):
-    if builder.format in ('html', 'slides'):
-        image_format = builder.config.blockdiag_html_image_format.upper()
-    elif builder.format == 'latex':
-        if builder.config.blockdiag_tex_image_format:
-            image_format = builder.config.blockdiag_tex_image_format.upper()
-        else:
-            image_format = builder.config.blockdiag_latex_image_format.upper()
+    config = builder.config
+    image_formats = config.blockdiag_image_format or {}
+
+    # Find image format in a configured dictionary; or assume that the
+    # provided configuration is a string format (for convenience).
+    if isinstance(image_formats, dict):
+        image_format = image_formats.get(builder.format)
+    else:
+        image_format = image_formats
+
+    # If a format is not registered by a user, check if any deprecated
+    # options are set.
+    if not image_format:
+        if builder.format in ('html', 'slides'):
+            image_format = config.blockdiag_html_image_format
+        elif builder.format == 'latex':
+            if config.blockdiag_latex_image_format:
+                image_format = config.blockdiag_latex_image_format
+            else:
+                image_format = config.blockdiag_tex_image_format
+
+    # If an explicit image format is not set, check if the wildcard
+    # entry defines a format.
+    if not image_format:
+        image_format = image_formats.get('*')
```

Comment:
This value should be uppercased.

Reply:
I believe the desire for uppercasing the image format before use is now enforced in the else case before we validate accepted format type (instead of trying to call `upper()` in multiple places).
```
...
    else:
        image_format = image_format.upper()
...
```